### PR TITLE
adding category: tag to categories and migration"

### LINKED ISSUE #153: Namespace categories better
Switch from `books` to `is:books`.

### PR DESCRIPTION
**Pull request by georgeteo**
I think this should fix #153 . 

I didn't change the lookup at all, but I don't think that's necessary, but please double check. 

Am afraid of breaking things badly if the cronjob runs and something doesnt work.

### CODE REVIEW

**Comment by fatlotus** [comment]
Looks good to me!

**Comment by georgeteo** [comment]
Your other ?

**Comment by georgeteo** [comment]
I think I fixed your comment. Please take a look again.

**Comment by georgeteo** [comment]
Bump?

File: caravel/storage/entities.py [hunk 1]
```diff
@@ -185,3 +185,8 @@ def recompute_admin_keys(listing):
     if not listing.admin_key:
         listing.admin_key = hashlib.sha1(caravel.app.secret_key +
             ":" + listing.key().name()).hexdigest()
+
+@Listing.migration(to_version=9)
```
Comment (fatlotus) [inline caravel/storage/entities.py]: I think you want to_version=10.

File: caravel/storage/entities.py [hunk 2]
```diff
@@ -126,10 +126,10 @@ def keywords(self):
 
         # Tokenize title and body (ranking them equally)
         words = [self.seller] + self.title.split() + self.body.split()
-        words += self.categories
         if self.price == 0:
             words += ["price:free"]
-        singularized = [fold_query_term(word) for word in words]
+        singularized = [fold_query_term(word) for word in words if word not in self.CATEGORIES_DICT.keys()]
```
Comment (fatlotus) [inline caravel/storage/entities.py]: Consider:
```
words = set(words) - set(self.CATEGORIES_DICT.keys())
singularized = [fold_query_term(word) for word in words + self.categories]
```

Comment (georgeteo) [inline caravel/storage/entities.py]: Fixed.

File: caravel/storage/entities.py (hunk 1 above)
Comment (georgeteo) [inline caravel/storage/entities.py]: Fixed.

**Comment by fatlotus** [comment]
oh dang everythings i done borked it. Might be worth reset'ing and then rebase'ing

**Comment by georgeteo** [comment]
Yes. Is totally wonky. It's like 5 lines of code. I might just make a new branch.

**Comment by georgeteo** [comment]
Although that wont change the test failing.

**Comment by fatlotus** [comment]
Keep working on it — if you can't figure it out, merge my PR and then try again with the new testing framework.

**Comment by georgeteo** [comment]
Thanks for the fix :)

**Comment by georgeteo** [comment]
OK. Fixed.

**Comment by georgeteo** [comment]
Wait.
Search doesnt work.

**Pull request merged**